runtime/lua/vim/lsp/sync.lua

```diff
@@ -59,31 +58,14 @@ local function byte_to_utf(line, byte, offset_encoding)
   -- convert to 0 based indexing for str_utfindex
   byte = byte - 1
 
-  local utf_idx
-  local _
-  -- Convert the byte range to utf-{8,16,32} and convert 1-based (lua) indexing to 0-based
-  if offset_encoding == 'utf-16' then
-    _, utf_idx = str_utfindex(line, byte)
-  elseif offset_encoding == 'utf-32' then
-    utf_idx, _ = str_utfindex(line, byte)
-  else
-    utf_idx = byte
-  end
+  local utf_idx = vim.lsp.util.str_utfindex_enc(line, byte, offset_encoding or vim.lsp.protocol.UTF8)
```

Comment:
Wanted to preserve that UTF-8 is the default here, let me know if this was intentional?